eox_nelp/programs/api/v1/views.py

```diff
@@ -34,6 +46,39 @@ def wrapper(self, request, *args, **kwargs):
     return decorator
 
 
+def require_national_id_query_param():
+    """Decorator to check if national_id query parameter is valid if provided.
+    If not provided, the view will proceed without filtering by national_id.
+    Using the request user if not provided.
+    Returns 400 if missing and 422 if invalid.
+    """
+    def decorator(func):
+        @wraps(func)
+        def wrapper(self, request, *args, **kwargs):
+            if self.request.query_params:
+                if not (national_id := self.request.query_params.get("national_id", "")):
+                    return Response(
+                        {
+                            "error": "MISSING_NATIONAL_ID",
+                            "message": "national_id query parameter is required."
+                        },
+                        status=status.HTTP_400_BAD_REQUEST,
+                    )
```

Comment:
I don't understand this block of code
if I make `GET /eox-nelp/api/programs/v1/program-lookup/ `is ok but if I make `GET /eox-nelp/api/programs/v1/program-lookup/?course_id=blablabla ` the national id is required ?? if so, why ?